src/apps/common/submodules/mediaSelect/mediaSelect.js

```diff
@@ -113,32 +120,44 @@ define(function(require) {
 			} else if (skin === 'tabs') {
 				template = $(self.getTemplate({
 					name: 'tabs-layout',
-					data: formattedData,
+					data: _.merge(formattedData, {
+						enableTextspeechTab: enableTextspeechTab
+					}),
 					submodule: 'mediaSelect'
 				}));
+
+				monster.ui.charsRemaining(template.find('.custom-greeting-text'), {
+					size: 350,
+					customClass: 'chars-remaining-counter'
+				});
+
 				self.mediaSelectBindTabsTemplate(template);
 			}
 
 			return template;
 		},
 
-		mediaSelectGetValue: function(template) {
+		mediaSelectGetValue: function(template, args, callback) {
 			var self = this,
-				response;
+				ttsTab = template.find('.monster-tab-content.monster-tab-content-tts.active');
 
 			if (template) {
 				var val = template.find('.media-dropdown').val();
 
 				if (val === 'shoutcast') {
-					response = template.find('.shoutcast-div input').val();
+					callback(template.find('.shoutcast-div input').val());
+				} else if (ttsTab.length) {
+					args.tts.text = ttsTab.find('.custom-greeting-text').val();
+
+					self.mediaSelectUpdateTTSMedia(args, function(id) {
+						callback(id);
+					});
```

Comment:
The callback function could be sent as a parameter as is:
```suggestion
					self.mediaSelectUpdateTTSMedia(args, callback);
```